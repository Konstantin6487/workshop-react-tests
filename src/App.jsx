import React from 'react';

const App = ({ name }) => (
  <div>
    Hello,
    {' '}
    {name}
  </div>
);

export default App;
