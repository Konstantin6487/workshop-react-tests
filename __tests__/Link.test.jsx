import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import App from '../src/App';

test('App', () => {
  const component = renderer.create(<App name="Sergey" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('allows us to set props', () => {
  const wrapper = mount(<App name="Ivan" />);
  expect(wrapper.render()).toMatchSnapshot();
});

it('renders three <Foo /> components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('div').length).toEqual(1);
});
