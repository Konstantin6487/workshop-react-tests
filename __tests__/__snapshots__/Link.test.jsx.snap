// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<div>
  Hello,
   
  Sergey
</div>
`;

exports[`allows us to set props 1`] = `
<div>
  Hello, Ivan
</div>
`;
